libraries/AP_WheelEncoder/WheelEncoder_SITL_Quadrature.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+#include "WheelEncoder_SITL_Quadrature.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_WheelEncoder_SITL_Qaudrature::AP_WheelEncoder_SITL_Qaudrature(AP_WheelEncoder &frontend, uint8_t instance, AP_WheelEncoder::WheelEncoder_State &state) :
+    AP_WheelEncoder_Backend(frontend, instance, state),
+    _sitl(AP::sitl())
+{
+}
+
+void AP_WheelEncoder_SITL_Qaudrature::update(void)
+{
+
+    // earth frame velocity of vehicle in vector form
+    Vector2f ef_velocity(_sitl->state.speedN, _sitl->state.speedE);
+
+    // store current heading
+    double current_heading = _sitl->state.yawDeg;
+
+    // transform ef_velocity vector to last_heading frame
+    // helps to calculate direction of movement
+    Vector2f vf_velocity;
+    vf_velocity.x = ef_velocity.x*cos(radians(current_heading)) + ef_velocity.y*sin(radians(current_heading));
+    vf_velocity.y = -ef_velocity.x*sin(radians(current_heading)) + ef_velocity.y*cos(radians(current_heading));
+
+    // calculate dt
+    uint32_t time_now = AP_HAL::millis();
+    double dt = (time_now - _state.last_reading_ms)/1000.0f;
+    if (is_zero(dt)) // sanity check
+        return;
+
+    // calculate turn rate from heading_diff
+    double turn_rate = radians(_sitl->state.yawRate);
+
+    double speed = norm(vf_velocity.x, vf_velocity.y);
+
+    if (vf_velocity.x < 0.0f) {
+        speed *= -1;
+    }
+
+    // distance from center of wheel axis to each wheel
+    double l = ( fabsf(_frontend.get_pos_offset(0).y) + fabsf(_frontend.get_pos_offset(1).y) )/2.0f;
+
+    if (_state.instance == 0) { 
+        speed = speed - turn_rate*l; // for left wheel
+    } else if(_state.instance ==1) {
+        speed = speed + turn_rate*l; // for right wheel
+    } else {
+        return; // invalid instance
+    }
+
+    double radius = _frontend.get_wheel_radius(_state.instance);
+    if (is_zero(radius)) // avoid divide by zero
```

Comment:
Add braces for this if's block.

Also... maybe `!is_positive?` instead?  (likewise above)

Reply:
... also, this is SITL, we can be nice and spit a warning at the user :-)